src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientLSBatchOpsImpl.java

```diff
@@ -0,0 +1,509 @@
+import org.slf4j.Logger;
+
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.alipay.oceanbase.rpc.util.TableClientLoggerFactory.LCD;
+import static com.alipay.oceanbase.rpc.util.TableClientLoggerFactory.RUNTIME;
+
+public class ObTableClientLSBatchOpsImpl extends AbstractTableBatchOps {
+
+    private static final Logger logger = TableClientLoggerFactory.getLogger(ObTableClientBatchOpsImpl.class);
+    private final ObTableClient obTableClient;
+    private ExecutorService executorService;
+    private boolean returningAffectedEntity = false;
+    private List<ObTableSingleOp> batchOperation;
+
+    /*
+     * Ob table client batch ops impl.
+     */
+    public ObTableClientLSBatchOpsImpl(String tableName, ObTableClient obTableClient) {
+        this.tableName = tableName;
+        this.obTableClient = obTableClient;
+        this.executorService = obTableClient.getRuntimeBatchExecutor();
+        this.batchOperation = new ArrayList<>();
+    }
+
+    /*
+     * Get ob table batch operation.
+     */
+    @Override
+    public ObTableBatchOperation getObTableBatchOperation() {
+        return null;
+    }
+
+    public List<ObTableSingleOp> getSingleOperations() {
+        return batchOperation;
+    }
+
+    /*
+     * Get.
+     */
+    @Override
+    public void get(Object[] rowkeys, String[] columns) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Update.
+     */
+    @Override
+    public void update(Object[] rowkeys, String[] columns, Object[] values) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Delete.
+     */
+    @Override
+    public void delete(Object[] rowkeys) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Insert.
+     */
+    @Override
+    public void insert(Object[] rowkeys, String[] columns, Object[] values) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Replace.
+     */
+    @Override
+    public void replace(Object[] rowkeys, String[] columns, Object[] values) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Insert or update.
+     */
+    @Override
+    public void insertOrUpdate(Object[] rowkeys, String[] columns, Object[] values) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Increment.
+     */
+    @Override
+    public void increment(Object[] rowkeys, String[] columns, Object[] values, boolean withResult) {
+        throw new FeatureNotSupportedException();
+    }
+
+    /*
+     * Append.
+     */
+    @Override
+    public void append(Object[] rowkeys, String[] columns, Object[] values, boolean withResult) {
+        throw new FeatureNotSupportedException();
+    }
+
+    private void addOperation(ObTableSingleOp singleOp) {
+        batchOperation.add(singleOp);
+    }
+
+    public void addOperation(CheckAndInsUp checkAndInsUp) {
```

Comment:
Should we use addOperation(CheckAndExecute xxx) and CheckAndInsUp(xxx) (just like other operations)